thread_safe_function_counting_wrapped/node-addon-api/addon.cc

```diff
@@ -0,0 +1,158 @@
+#include <chrono>
+#include <functional>
+#include <thread>
+
+#include "napi.h"
+
+template <class T>
+struct ExtractJSFunctionImpl;
+
+class NodeJSContext final {
+ private:
+  struct Impl final {
+    Napi::Env env_;
```

Comment:
We really don't recommend storing the `Napi::Env`. You can access it from `promise_.Env()`

Reply:
Good point, thanks, https://github.com/C5T/Current/issues/904.